src/menu/cheat_load.c

```diff
@@ -0,0 +1,197 @@
+		i++;
+	}
+	free(tab);
+	return (NULL);
+}
+
+char	**ft_split(char const *s, char c)
+{
+	char	**tab;
+	char	*next_str;
+	size_t	next_strlen;
+	int		i;
+
+	i = -1;
+	if (!s)
+		return (NULL);
+	tab = malloc(sizeof(char *) * (find_str(s, c) + 1));
+	if (!tab)
+		return (NULL);
+	next_str = (char *)s;
+	next_strlen = 0;
+	while (++i < find_str(s, c))
+	{
+		get_next_str(&next_str, &next_strlen, c);
+		tab[i] = (char *)malloc(sizeof(char) * (next_strlen + 1));
+		if (!tab[i])
+			return (free_tab(tab));
+		strlcpy(tab[i], next_str, next_strlen + 1);
+	}
+	tab[i] = NULL;
+	return (tab);
+}
+
+cheat_load_err_t load_cheats(menu_t *menu) {
+    FILE *cheatsFile;
+    struct stat st;
+    size_t cheatsLength;
+	path_t *path = path_clone(menu->load.rom_path);
+
+    //Parse cheats from file
+    path_ext_replace(path, "cht");
+    if((cheatsFile = fopen(path_get(path), "rb")) == NULL) {
+		path_free(path);
+        return CHEAT_LOAD_OK;//no file is not an error.
+    }
+
+    if (fstat(fileno(cheatsFile), &st)){
+		path_free(path);
+        return CHEAT_LOAD_ERR_SIZE_FAILED;
+    }
+
+    cheatsLength = st.st_size;
+    if (cheatsLength <= 0) {
+	   path_free(path);
+        return CHEAT_LOAD_ERR_CHEAT_EMPTY;
+    }
+    if (cheatsLength > KiB(128)) {
+		path_free(path);
+        return CHEAT_LOAD_ERR_CHEAT_TOO_LARGE;
+    }
+
+    char *cheatsContent = NULL;
+    if((cheatsContent = malloc((cheatsLength + 1) * sizeof(char))) == NULL) {
+		path_free(path);
+        return CHEAT_LOAD_ERR_MALLOC_FAILED;
+    }
+    if(fread(cheatsContent, cheatsLength, 1, cheatsFile) != 1) {
+	   	path_free(path);
+        return CHEAT_LOAD_ERR_READ_FAILED;
+    }
+
+    cheatsContent[cheatsLength] = '\0';
+    if(fclose(cheatsFile) != 0){
+		path_free(path);
+        return CHEAT_LOAD_ERR_CLOSE_FAILED;
+    }
+    cheatsFile = NULL;
+ 
+    char **tab = ft_split(cheatsContent, '\n');
+	size_t lines = 1;
+	for (size_t i = 0; tab[i] != NULL; i++) {
+		lines++;
+	}
+
+    free(cheatsContent);
+
+    uint32_t  *cheats = (uint32_t*)malloc(((lines * sizeof(uint32_t)) * 2) + 2);
+	memset(cheats, 0, ((lines * sizeof(uint32_t)) * 2) + 2);
+    size_t cheatIndex = 0;
+    for(size_t i = 0; tab[i] != NULL; i++) {
+        //ignore titles 
+        if (tab[i][0] == '#') {
+            continue;
+        }
+		//ignore empty, too small or too big lines
+        if (strlen(tab[i]) < 12 || strlen(tab[i]) > 15) {
+            continue;
+        }
+        char **splitCheat = ft_split(tab[i], ' ');
+		//thank you mena for checking my fucky wucky
+        uint32_t cheatValue1 = strtoul(splitCheat[0], NULL, 16);
+        uint32_t cheatValue2 = strtoul(splitCheat[1], NULL, 16);
+        cheats[cheatIndex] = cheatValue1;
+        cheats[cheatIndex + 1] = cheatValue2;
+        free_tab(splitCheat);
+        cheatIndex += 2;
+    }
+    free_tab(tab);
+
+    cheats[cheatIndex] = 0;
+    cheats[cheatIndex + 1] = 0;
+    menu->boot_params->cheat_list = cheats;
+    return CHEAT_LOAD_OK;
+}
```

Comment:
```suggestion
}

```
Add an EOF line for GH